CMakeLists.txt

```diff
@@ -132,6 +132,35 @@ if(CUDA_FOUND)
   include_directories(${CUDA_INCLUDE_DIRS})
 endif()
 
+# include FPGA target only when CMAKE_ENABLE_FPGA is set to true
+if(CMAKE_ENABLE_FPGA)
+  find_file(CL_HPP CL/opencl.h)
+  if(NOT CL_HPP)
+    set(HAVE_FPGA 0)
+    message(WARNING " OpenCL headers were not found, deactivating FPGA target.")
+  else()
+    set(HAVE_FPGA 1)
+    add_compile_options(-DPORTS_CONFIG=PORTS_2r_3r3w_ddr)  # selects the kernel's memory ports configuration; must be in sync with available kernel bitstream
+    add_compile_options(-DBDA_DEBUG_LEVEL=0)  # set debug messages level for FPGA library functions (should be set to 0 for Release build)
+    # include directories for the FPGA library
+    include_directories(${_modules_dir}/opm-simulators/opm/simulators/linalg/bda/fpga/src/sda_app)
+    include_directories(${_modules_dir}/opm-simulators/opm/simulators/linalg/bda/fpga/src/sda_app/common)
+    # add external project to compile the FPGA library
+    include (ExternalProject)
+    ExternalProject_Add(FPGA_library
+      SOURCE_DIR ${_modules_dir}/opm-simulators/opm/simulators/linalg/bda/fpga/alveo_u280/vitis_20192/OPM-integration
+      BUILD_IN_SOURCE 1
+      # force the build step to always be run because source dependencies cannot be made explicit
+      BUILD_ALWAYS 1
```

Comment:
This is needed to always execute the makefile, which in turn will trigger compilation only when any library file has been modified (thus it should not significantly impact the build time).
Since cmake does not know which files are used by the external project, this seems to be the only way to trigger an update if they are changed.